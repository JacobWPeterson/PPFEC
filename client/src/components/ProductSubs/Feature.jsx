import React from 'react';
import styled from 'styled-components';

const ListItem = styled.li`
  &:before { content: 'âœ“';
    color: #80ccc4;
    padding-right: .5rem; };
  color: #424242;
`;

const Feature = ({ feature }) => {
  if (feature.value) {
    return (
      <ListItem>{`${feature.feature}: ${feature.value}`}</ListItem>
    );
  }
  return (
    <ListItem>{feature.feature}</ListItem>
  );
};

export default Feature;
