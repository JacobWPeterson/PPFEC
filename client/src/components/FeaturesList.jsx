import React from 'react';
import Feature from './Feature.jsx';
import styled from 'styled-components';

const List = styled.ul`
  list-style: none;
  flex-basis: 40%;
`

const FeaturesList = ({ features }) => (
  <List>
    {features.map((feature, index) => <Feature key={index} feature={feature} />)}
  </List>
)

export default FeaturesList;
