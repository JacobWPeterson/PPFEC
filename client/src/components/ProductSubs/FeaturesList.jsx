import React from 'react';
import styled from 'styled-components';
import Feature from './Feature.jsx';

const List = styled.ul`
  list-style: none;
  flex-basis: 33%;
  padding-left: 1.5vw;
`;

const FeaturesList = ({ features }) => (
  <List>
    {features.map((feature, index) => <Feature key={index} feature={feature} />)}
  </List>
);

export default FeaturesList;
