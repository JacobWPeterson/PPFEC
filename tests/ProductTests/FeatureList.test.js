/* eslint-disable no-undef */
/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { mount } from 'enzyme';
import styled from 'styled-components';

import FeaturesList from '../../client/src/components/ProductSubs/FeaturesList.jsx';
import Feature from '../../client/src/components/ProductSubs/Feature.jsx';

const List = styled.ul`
  list-style: none;
  flex-basis: 33%;
  padding-left: 1.5vw;
`;

const props = {
  features: [{ feature: "Sole", value: "Rubber" }, {feature: "Mid-Sole", value: "ControlSupport Arch Bridge"}, {feature: "Material", value: null}],
};

describe('FeaturesList', () => {
  const FeaturesListComponent = mount(<FeaturesList {...props} debug />);

  it('Should render correctly', () => {
    expect(FeaturesListComponent).toMatchSnapshot();
  });

  it('renders the feature list', () => {
    expect(FeaturesListComponent.find(Feature).length).toEqual(3);
  });
});
