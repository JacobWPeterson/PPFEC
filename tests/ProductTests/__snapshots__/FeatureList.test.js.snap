// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturesList Should render correctly 1`] = `
<FeaturesList
  debug={true}
  features={
    Array [
      Object {
        "feature": "Sole",
        "value": "Rubber",
      },
      Object {
        "feature": "Mid-Sole",
        "value": "ControlSupport Arch Bridge",
      },
      Object {
        "feature": "Material",
        "value": null,
      },
    ]
  }
>
  <FeaturesList__List>
    <ul
      className="FeaturesList__List-btq5gp-0 gsMclF"
    >
      <Feature
        feature={
          Object {
            "feature": "Sole",
            "value": "Rubber",
          }
        }
        key="undefined + Sole"
      >
        <Feature__ListItem>
          <li
            className="Feature__ListItem-sc-2j09xg-0 kZXsUK"
          >
            Sole: Rubber
          </li>
        </Feature__ListItem>
      </Feature>
      <Feature
        feature={
          Object {
            "feature": "Mid-Sole",
            "value": "ControlSupport Arch Bridge",
          }
        }
        key="undefined + Mid-Sole"
      >
        <Feature__ListItem>
          <li
            className="Feature__ListItem-sc-2j09xg-0 kZXsUK"
          >
            Mid-Sole: ControlSupport Arch Bridge
          </li>
        </Feature__ListItem>
      </Feature>
      <Feature
        feature={
          Object {
            "feature": "Material",
            "value": null,
          }
        }
        key="undefined + Material"
      >
        <Feature__ListItem>
          <li
            className="Feature__ListItem-sc-2j09xg-0 kZXsUK"
          >
            Material
          </li>
        </Feature__ListItem>
      </Feature>
    </ul>
  </FeaturesList__List>
</FeaturesList>
`;
